sph/include/sph/eos.hpp

```diff
@@ -9,6 +9,33 @@ namespace sphexa
 namespace sph
 {
 
+template<class T1, class T2>
+CUDA_DEVICE_HOST_FUN util::tuple<T2, T2> equationOfState(T1 u, T2 rho)
+{
+    constexpr T2 gamma = (5.0 / 3.0);
+
+    T2 tmp = u * (gamma - T1(1));
+    T2 p   = rho * tmp;
+    T2 c   = std::sqrt(tmp);
+
+    return {p, c};
+}
+
+template<class T1, class T2, class T3>
+CUDA_DEVICE_HOST_FUN util::tuple<T2, T2> equationOfState(T1 u, T2 rho, T3 mui)
+{
+    constexpr T2 R     = 8.317e7;
+    constexpr T2 gamma = (5.0 / 3.0);
+
+    T2 cv   = T2(1.5) * R / mui;
+    T2 temp = u / cv;
+    T2 tmp  = u * (gamma - T1(1));
+    T2 p    = rho * tmp;
+    T2 c    = std::sqrt(tmp);
+
+    return {p, c, cv, temp};
+}
+
```

Comment:
These two EOS are named the same. How we differentiate between them at the time of choosing one?

Reply:
They are the same EOS, but one computes additional quantities. The differentiation occurs by the signature, i.e. different number of arguments.